config.go

```diff
@@ -43,7 +43,9 @@ type SpanNameFormatter interface {
 // AttributesGetter provides additional attributes on spans creation.
 type AttributesGetter func(ctx context.Context, method Method, query string, args []driver.NamedValue) []attribute.KeyValue
 
-type config struct {
+type SpanOmitter func(ctx context.Context, cfg Config, method Method, enableDBStatement bool, query string, args []driver.NamedValue) bool
```

Comment:
> Any reason for adding these parameters?

I was just adding everything that was available without much thought. If you don't think they're useful I can remove.  The signature is quite long as it is.

> Do you think the new name is a good idea?

No problem about the naming. Changes like this are easy to make.  I found myself not liking SpanOmitter because it sounds like SpanEmitter which is sort of the opposite meaning.  Although sometimes I have similar confusion around the word Filter.  Are we filtering to remove? or filtering to keep? It's not obvious from the name but docs should suffice.

FWIW, filter seems to line up with the otel nomenclature, although they further specify inclusion vs exclusion which makes it clear which is happening.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/filterprocessor/README.md

Let me know which you want it to be.

Reply:
also, not passing cfg means I don't need to export it which is nice

Reply:
> I was just adding everything that was available without much thought. If you don't think they're useful I can remove. The signature is quite long as it is.

Yeah, let's remove these parameters.

>  Although sometimes I have similar confusion around the word Filter. Are we filtering to remove? or filtering to keep? It's not obvious from the name but docs should suffice.

I had the same concern as you when I entered the "filter" world. But I checked some well-known implementations of the filter function, and they all use the filtering-to-keep approach. I guess this approach has become a convention.

- https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.filter.html
- https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html
- https://www.typescriptlang.org/docs/handbook/2/narrowing.html
- https://ericniebler.github.io/range-v3/structranges_1_1views_1_1filter__fn.html

> FWIW, filter seems to line up with the otel nomenclature, although they further specify inclusion vs exclusion which makes it clear which is happening.

Yes, it is more clear with inclusion and exclusion. It is also worth mentioning that an otel `contrib` project is a collection of 3rd-party packages. It doesn't define the otel specification. For formal otel specification, you need to check https://github.com/open-telemetry/opentelemetry-specification

Reply:
works for me.  I've made the change.  Let me know if I missed anything.